docs/spec/spec-proposals/f1-fee-distribution/f1_fee_distr.tex

```diff
@@ -68,12 +88,12 @@ \subsection{Commission Rates}
 
 In block $h$ a validator receives $f_h$ fees.
 Instead of incrementing that validators ``total accrued fees this period variable" by $f_h$, it is instead incremented by $(1 - commission\_rate) * f_p$.
-Then $commission\_rate * f_p$ is deposited directly to the validator.
-This scheme allow for updates to a validator's commission rate every block if desired.
+Then $commission\_rate * f_p$ is deposited directly to the validator's account.
+This allows for efficient updates to a validator's commission rate every block if desired.
 
 \subsection{Slashing}
 \label{ssec:slashing}
-Slashing is distinct from withdrawals, since not only does it lower the validators total amount of stake, but it also lowers each of its delegator's stake by a fixed percentage.
+Slashing is distinct from withdrawals, since lowers the stake of all of the delegator's by a fixed percentage.
 Since noone is charged gas for slashes, a slash cannot iterate over all delegators.
 Thus we can no longer just multiply by $x$ over the difference in stake. 
 The solution here is to instead store each period created by a slash in the validators state.
```

Comment:
This seems fine for the case of delegators getting slashed, then withdrawing fees. It's still the case that for delegators which have started unbonding or redelegating their stake before evidence is discovered (for which they are liable), we will need to iterate over the unbonding delegations / redelegations and decrease their stake (in the former case) or decrease their stake in the redelegated-to validator as if unbonding (in the latter case).